plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftPvcHelper.java

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.che.plugin.openshift.client;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaimVolumeSource;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaimVolumeSourceBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+/**
+ * Helper class for executing simple commands in a Persistent Volume on Openshift.
+ * <p>
+ * Creates a short-lived Pod using a CentOS image which mounts a specified PVC and
+ * executes a command (either {@code mkdir -p <path>} or {@code rm -rf <path}). Reports
+ * back whether the pod succeeded or failed. Supports multiple paths for one command.
+ * <p>
+ * For mkdir commands, an in-memory list of created workspaces is stored and used to avoid
+ * calling mkdir unnecessarily. However, this list is not persisted, so dir creation is
+ * not tracked between restarts.
+ *
+ * @author amisevsk
+ */
+final class OpenShiftPvcHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpenShiftPvcHelper.class);
+
+    private static final String POD_PHASE_SUCCEEDED        = "Succeeded";
+    private static final String POD_PHASE_FAILED           = "Failed";
+    private static final String JOB_IMAGE                  = "centos:centos7";
```

Comment:
Done. The image used is now specified by the property `che.openshift.jobs.image`, which is injected into the `OpenShiftWorkspaceFilesCleaner` and `OpenShiftConnector`. Using the workspace's image won't necessarily work, since delete jobs are executed when the server is idled, so there's no workspace image string readily available.